Source/CesiumRuntime/Private/Tests/GlobeAwareDefaultPawn.spec.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2023 CesiumGS, Inc. and Contributors
+
+#if WITH_EDITOR
+
+#include "CesiumGlobeAnchorComponent.h"
+#include "CesiumTestHelpers.h"
+#include "Editor.h"
+#include "Engine/World.h"
+#include "EngineUtils.h"
+#include "GlobeAwareDefaultPawn.h"
+#include "Misc/AutomationTest.h"
+#include "Tests/AutomationEditorCommon.h"
+
+BEGIN_DEFINE_SPEC(
+    FGlobeAwareDefaultPawnSpec,
+    "Cesium.GlobeAwareDefaultPawn",
+    EAutomationTestFlags::ApplicationContextMask |
+        EAutomationTestFlags::ProductFilter)
+
+FDelegateHandle subscriptionPostPIEStarted;
+
+END_DEFINE_SPEC(FGlobeAwareDefaultPawnSpec)
+
+void FGlobeAwareDefaultPawnSpec::Define() {
+  Describe(
+      "should not spike altitude when very close to final destination",
```

Comment:
```suggestion
      "AGlobeAwareDefaultPawn",
```

Reply:
I intentionally didn't do it this way...

What you've proposed would be fine for right now. But it would break down the moment we add the next test. We can't add another `It` to the existing `Describe`, because then all the same `BeforeEach` and `AfterEach` actions would run for that new test as well. So we'd have to add a new `Describe`. What would we call the new one? `AGlobeAwareDefaultPawn2`? :)

It's kind of silly how the only way (AFAICT) to do a sequence of actions over frames is with these multiple `BeforeEach` actions. But it is what it is, and Epic themselves have run into the same problem, and they adopted the same pattern I did, which is of course where I got the idea. See here for example:
https://github.com/EpicGames/UnrealEngine/blob/5.2.1-release/Engine/Plugins/Tests/AutomationDriverTests/Source/AutomationDriverTests/Private/AutomationDriver.spec.cpp#L287

The test UI in the Editor handles this specially by hiding the anonymous test and making the one test have the label of the Describe instead. So it all works just fine, other than the kind of weird looking anoymous `It`.

Reply:
Sounds like a reasonable course of action.